api/config/v1alpha1/envoyproxy_types.go

```diff
@@ -164,6 +167,49 @@ const (
 	LogLevelError LogLevel = "error"
 )
 
+// EnvoyProxyTrafficSettings defines the traffic settings for requests being
+// processed by the EnvoyProxy data plane.
+type EnvoyProxyTrafficSettings struct {
+	// LoadBalancer defines the load balancer settings between
+	// the Envoy Proxy data plane and the upstream hosts.
+	//
+	// +optional
+	LoadBalancer *EnvoyProxyLoadBalancerSetting `json:"loadBalancer,omitempty"`
+}
+
+type EnvoyProxyLoadBalancerSetting struct {
+	// Strategy for load balancing.
+	//
+	// +kubebuilder:default=LeastRequest
+	Strategy LoadBalancerStrategy `json:"strategy,omitempty"`
+}
+
+// LoadBalancerStrategy defines the type of load balancing strategy to use.
+//
+// +kubebuilder:validation:Enum=LeastRequest;RoundRobin;Random;ClusterIP
+type LoadBalancerStrategy string
+
+const (
+	// LoadBalancerStrategyLeastRequest defines the "LeastRequest" load balancing strategy.
+	// In this mode, the upstream hosts is picked based on fewer active requests.
+	// Visit https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/load_balancers#weighted-least-request for more details.
+	LoadBalancerStrategyLeastRequest LoadBalancerStrategy = "LeastRequest"
+
+	// LoadBalancerStrategyRoundRobin defines the "RoundRobin" load balancing strategy.
+	// In this mode, the upstream host is selected in weighed round robin order.
+	// Visit https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/load_balancers#weighted-round-robin for more details.
+	LoadBalancerStrategyRoundRobin LoadBalancerStrategy = "RoundRobin"
+
+	// LoadBalancerStrategyRandom defines the "Random" load balancing strategy.
+	// In this mode, the upstream host is chosen randomly.
+	LoadBalancerStrategyRandom LoadBalancerStrategy = "Random"
+
+	// LoadBalancerStrategyClusterIP defines the "ClusterIP" load balancing strategy.
+	// In this mode, EnvoyProxy routes requests to the ClusterIP of the service and
+	// relies on another entity such as kube-proxy to perform load balancing to the upstream hosts.
+	LoadBalancerStrategyClusterIP LoadBalancerStrategy = "ClusterIP"
```

Comment:
does this mean, eds will just send ClusterIP?

Reply:
yes, this is what is happening as of v0.4.0

Reply:
correct me if I'm wrong, only `ClusterIP` will change the result of `EDS`, other types only change `LbPolicy` on envoy cluster?

Reply:
yeah thats right :)

Reply:
this's a little odd to me, Is there any way to make them more consistent?

Reply:
can you share why its odd, this API is meant for end users (platform owners / app owners) to configure LB policy, so they can
* Use LB Policies provided at a L7 request level by Envoy (round-robin, least request, etc) or
* LB Policies provided at a L3/L4 flow level by another intermediate data plane entity such as kube-proxy